v2/spanner-to-sourcedb/terraform/samples/spanner-to-sharded-mysql/README.md

```diff
@@ -0,0 +1,508 @@
+an already executing job.
+
+Look for the following log during `terraform apply` -
+
+```shell
+  # google_dataflow_flex_template_job.reverse_replication_job will be updated in-place
+```
+
+### Updating workers of a Dataflow job
+
+Currently, the Terraform `google_dataflow_flex_template_job` resource does not
+support updating the workers of a Dataflow job.
+If the worker counts are changed in `tfvars` and a Terraform apply is run,
+Terraform will attempt to cancel/drain the existing Dataflow job and replace it
+with a new one.
+**This is not recommended**. Instead, use the `gcloud` CLI to update the worker
+counts of a launched Dataflow job.
+
+```shell
+gcloud dataflow jobs update-options \                                                                                                                                                                                                                                                                                                                      (base) 
+        --region=us-central1 \
+        --min-num-workers=5 \
+        --max-num-workers=20 \
+      2024-06-17_01_21_44-12198433486526363702
+```
+
+### Long time to delete SMT bucket
+
+A GCS bucket can only be deleted if all its comprising objects are deleted
+first. This template attempts to delete all objects in the created GCS bucket
+and then deletes the bucket as well.
+**This may take a long time, depending on the size of the data in the bucket.**
+If you want Terraform to only create the GCS bucket but skip its deletion
+during `terraform destroy`, you will have to use the `terraform state rm` API to
+delete GCS resource from being traced by Terraform after the `apply` command.
+
+### Source shard configuration file
+
+Source shard configuration file that is supplied to the dataflow is automatically
+created by Terraform. A sample file that this uploaded to GCS looks like
+below - 
+
+```json
+[
+    {
+    "logicalShardId": "shard1",
+    "host": "10.11.12.13",
+    "user": "root",
+    "secretManagerUri":"projects/123/secrets/rev-cmek-cred-shard1/versions/latest",
+    "port": "3306",
+    "dbName": "db1"
+    },
+    {
+    "logicalShardId": "shard2",
+    "host": "10.11.12.14",
+    "user": "root",
+    "secretManagerUri":"projects/123/secrets/rev-cmek-cred-shard2/versions/latest",
+    "port": "3306",
+    "dbName": "db2"
+    }
+]
+```
+
+This file is generated from the configuration provided in the `var.shard_list`
+parameter.
+
+### Adding access to Terraform service account
+
+#### Using custom role and granular permissions (recommended)
+
+You can run the following gcloud command to create a custom role in your GCP
+project.
+
+```shell
+gcloud iam roles create live_migrations_role --project=<YOUR-PROJECT-ID> --file=perms.yaml --quiet
+```
+
+The `YAML` file required for the above will be like so -
+
+```shell
+title: "Live Migrations Custom Role"
```

Comment:
Did not add that file since the permissions are documented in the README itself and the file could quickly become out of sync. Hence decided to keep all the permissions in only one place.